src/nupic/regions/tm_region.capnp

```diff
@@ -1,11 +1,17 @@
 @0xb9d11462f08c1dee;
 
+using import "/nupic/algorithms/backtracking_tm.capnp".BacktrackingTMProto;
+using import "/nupic/algorithms/backtracking_tm_cpp.capnp".BacktrackingTMCppProto;
 using import "/nupic/proto/TemporalMemoryProto.capnp".TemporalMemoryProto;
 
-# Next ID: 11
+# Next ID: 13
 struct TMRegionProto {
   temporalImp @0 :Text;
-  temporalMemory @1 :TemporalMemoryProto;
+  union {
+    temporalMemory @1 :TemporalMemoryProto;
+    backtrackingTM @11 :BacktrackingTMProto;
+    backtrackingTMCpp @12 :BacktrackingTMCppProto;
```

Comment:
We don't have to worry about renumbering because we aren't maintaining backwards compatibility. But there is no advantage to renumbering all of them. Actually I worry that perfectly number fields in all of our schema will lead people to think that that order must be maintained. So I could change it but not sure I'm achieving anything positive in doing that.